django/core/management/commands/makemessages.py

```diff
@@ -382,10 +383,15 @@ def handle(self, *args, **options):
             self.invoked_for_django = True
         else:
             if self.settings_available:
-                self.locale_paths.extend(settings.LOCALE_PATHS)
+                for path in settings.LOCALE_PATHS:
+                    locale_path = os.path.abspath(path)
+                    if locale_path not in self.locale_paths:
+                        self.locale_paths.append(locale_path)
             # Allow to run makemessages inside an app dir
             if os.path.isdir("locale"):
-                self.locale_paths.append(os.path.abspath("locale"))
+                locale_path = os.path.abspath("locale")
+                if locale_path not in self.locale_paths:
+                    self.locale_paths.append(locale_path)
```

Comment:
I think you mixed this patch with some unrelated changes in `makemessages`, please remove them.